### PR TITLE
[tests] record gas usage and events as part of adapter transactional …

### PR DESCRIPTION
**Pull request by sblackshear**
…test

## Description 

- Show gas usage and events in every expected value result
- Use realistic gas price (1000) and storage gas price (76) values
- Allow ProtocolConfig to be defined on a per-test or per-function level
- What I would really like to (but can't figure out how yet) do is have expected outputs  of each test for each protocol version to make sure that changes in `main` don't accidentally affect the semantics of older protocol versions. @tnowacki may have clever ideas on this.

The motivation for this is increasing the probability of detecting backward-incompatible changes. This seems particularly useful w.r.t gas--we basically get a lot of gas compatibility tests "for free".

## Test Plan 

These are tests.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-transactional-test-runner/src/args.rs
Comment (tnowacki) [inline crates/sui-transactional-test-runner/src/args.rs:46]: I love it, thanks for doing this! Was on someones radar, maybe @tzakian or @rvantonder?

**Review by tnowacki — COMMENTED** [review]
> What I would really like to (but can't figure out how yet) do is have expected outputs of each test for each protocol version to make sure that changes in main don't accidentally affect the semantics of older protocol versions. @tnowacki may have clever ideas on this.

The easiest thing to do, if we don't mind these tests being really slow, is to just rerun all of them with a separate `.exp` file for each protocol version. 
And maybe we could do something were we only print a separate `.exp` file if it is different than the current max protocol version?

**Comment by sblackshear** [comment]
> The easiest thing to do, if we don't mind these tests being really slow, is to just rerun all of them with a separate `.exp` file for each protocol version. And maybe we could do something were we only print a separate `.exp` file if it is different than the current max protocol version?

Like this idea--is it straightforward to hack the runner to do this? I couldn't quite figure out how because it's being driven by a datatest macro at the top level

**Comment by tnowacki** [comment]
>

> Like this idea--is it straightforward to hack the runner to do this? I couldn't quite figure out how because it's being driven by a datatest macro at the top level

It can be, though it would be a bit funky with the `init` flag for protocol config. 

The easiest way to do it would be just to add a small loop for each test, which means that each version of each test would run sequentially. More complicated, we could modify the top level macro that creates the tests. 
Probably best to just add a small loop for each test... though this will balloon the time for these tests by a lot. Shouldn't be a big deal besides the limit tests.

**Comment by dariorussi** [comment]
I agree with the multiple `exp` per protocol config, it would be very helpful!
Not clear about test failures.
In case @tnowacki may want to comment some more I'll leave it to him to accept

File: crates/sui-adapter-transactional-tests/tests/runtime_behavior/error_locations.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/runtime_behavior/error_locations.exp:14]: @tnowacki flagging this one as a potentially unexpected change (maybe we need a bigger budget with the new gas prices)?

File: crates/sui-adapter-transactional-tests/tests/size_limits/deleted_id_limits_tests.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/deleted_id_limits_tests.exp:13]: And these are definitely unexpected + I will fix.

File: crates/sui-adapter-transactional-tests/tests/size_limits/event_limits_tests.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/event_limits_tests.exp:14]: Will fix

File: crates/sui-adapter-transactional-tests/tests/size_limits/move_object_size_limit.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/move_object_size_limit.exp:12]: Will fix

File: crates/sui-adapter-transactional-tests/tests/size_limits/new_id_limits_tests.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/new_id_limits_tests.exp:13]: Will fix

File: crates/sui-adapter-transactional-tests/tests/size_limits/object_runtime_limits.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/object_runtime_limits.exp:17]: Will fix

File: crates/sui-adapter-transactional-tests/tests/size_limits/transfered_id_limits_tests.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/transfered_id_limits_tests.exp:14]: Will fix

File: crates/sui-adapter-transactional-tests/tests/size_limits/vector_len_limits.exp
Comment (sblackshear) [inline crates/sui-adapter-transactional-tests/tests/size_limits/vector_len_limits.exp:17]: Will fix

File: crates/sui-adapter-transactional-tests/tests/runtime_behavior/error_locations.exp
Comment (tnowacki) [reply crates/sui-adapter-transactional-tests/tests/runtime_behavior/error_locations.exp:14]: My guess would be it went from failing in a call, to somewhere much earlier (like when loading objects)

**Comment by sblackshear** [comment]
Fixed all unexpected changes in .exp files, this should be good to go

**Review by dariorussi — APPROVED** [review]
I am personally very happy with this and we can add extensions (I believe the multiple expected output is not there yet) at a later time.
So approving, @tnowacki if you had other things you wanted to see here please comment

**Pull request merged**
